crates/server/src/certificates.rs

```diff
@@ -0,0 +1,164 @@
+    let cert_file = certificate_dir.join("cert.pem");
+    // Read the certificate file
+    let cert_pem = fs::read_to_string(cert_file)?;
+    let (_, pem) = parse_x509_pem(cert_pem.as_bytes())?;
+    let x509_cert = pem.parse_x509()?;
+
+    // Get the local IP address
+    let local_ip = local_ip()?;
+
+    // Parse the local IP address to Vec<u8>
+    let local_ip_bytes = match local_ip {
+        std::net::IpAddr::V4(ip) => ip.octets().to_vec(),
+        std::net::IpAddr::V6(ip) => ip.octets().to_vec(),
+    };
+
+    // Flag for found IP address in certificate
+    let mut ip_found = false;
+
+    for ext in x509_cert.extensions() {
+      // Parse the extension
+      let parsed_ext = ext.parsed_extension();
+      
+      // Check if it's a SubjectAlternativeName extension
+      if let ParsedExtension::SubjectAlternativeName(san) = parsed_ext {
+          // Iterate over the general names
+          for gn in &san.general_names {
+              // Check if the general name is an IP address
+              if let GeneralName::IPAddress(dns_name) = gn {
+                  if *dns_name == local_ip_bytes {
+                      ip_found = true;
+                      break;
+                  }
+              }
+          }
+      }  
+    }
+
+    if !ip_found {
+        delete_certificate()?;
+        generate_certificate()?;
+    }
+
+    Ok(())
+}
+
+fn write_out_instructions() {
+    // Get full path to the certificate directory
+    let certificate_dir = get_certificate_dir();
+
+    println!("*******************************************************************************");
+    println!("To install the generated self-signed SSL certificate, follow these steps:");
+    println!("1. Navigate to {:?} directory", certificate_dir.to_str().unwrap());
+    println!("2. Depending on your operating system, the installation process will vary:");
+    println!("   - For macOS:");
+    println!("     a. Open Keychain Access and select the 'System' keychain.");
+    println!("     b. Drag the cert.pem file into Keychain Access.");
+    println!("     c. Double-click the certificate, expand the 'Trust' section, and set 'When using this certificate' to 'Always Trust'.");
+    println!("   - For Windows:");
+    println!("     a. Double-click the cert.pem file.");
+    println!("     b. Select 'Install Certificate...', choose the 'Local Machine' store location, and place it in the 'Trusted Root Certification Authorities' store.");
+    println!("   - For Linux:");
+    println!("     a. The process varies by distribution and browser. For most browsers, importing the certificate through the browser's settings by navigating to 'Privacy and Security' -> 'Manage Certificates' should work.");
+    println!("3. After installation, you may need to restart your browser or any application that needs to trust the certificate.");
+    println!("*******************************************************************************");    
```

Comment:
I think these should be part of the doc instead, and linked to by the CLI

https://github.com/calimero-network/calimero-network.github.io/pull/10/files#r1682828762

Otherwise, since this is running locally, we can know what platform it's executing on and only show instructions for that platform.

There's also the potential for this to run in a background process, `docker compose up` for example

Logs shouldn't include useful instructions (unless contextual from runtime knowledge), docs should.